### PR TITLE
Enable MIR inlinig for `#[inline(always)]` when mir-opt-level=1

### PR DESCRIPTION
**Pull request by eduardosm**
https://github.com/rust-lang/rust/issues/81567

### CODE REVIEW

File: compiler/rustc_mir_transform/src/inline.rs [hunk 1]
```diff
@@ -45,15 +45,7 @@ impl<'tcx> MirPass<'tcx> for Inline {
...
-        match sess.mir_opt_level() {
-            0 | 1 => false,
-            2 => {
-                (sess.opts.optimize == OptLevel::Default
-                    || sess.opts.optimize == OptLevel::Aggressive)
-                    && sess.opts.incremental == None
-            }
-            _ => true,
-        }
+        sess.mir_opt_level() > 0
```
Comment (oli-obk) [inline compiler/rustc_mir_transform/src/inline.rs:48]: We need to keep the incremental opt-out that existed for opt-level=2. So I think all that should be done here is to move the `| 1` to the second match arm.

**Comment by eduardosm** [comment]
Closing in favor of https://github.com/rust-lang/rust/pull/110560

**Pull request closed without merging**